src/main/kotlin/it/pagopa/wallet/eventdispatcher/api/WalletsApi.kt

```diff
@@ -0,0 +1,26 @@
+package it.pagopa.wallet.eventdispatcher.api
+
+import it.pagopa.generated.wallets.model.ClientId
+import it.pagopa.generated.wallets.model.UpdateWalletUsageRequest
+import java.time.OffsetDateTime
+import java.util.*
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+
+@Component
+class WalletsApi(private val walletsApiClient: it.pagopa.generated.wallets.api.WalletsApi) {
+
+    fun updateWalletUsage(
+        walletId: UUID,
+        clientId: ClientId,
+        usedAt: OffsetDateTime
+    ): Mono<ResponseEntity<Unit>> {
+        return walletsApiClient
+            .updateWalletUsageWithHttpInfo(
```

Comment:
generated api client uses [WebClient.retrieve()](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/reactive/function/client/WebClient.RequestHeadersSpec.html#retrieve()) method that, as per javadoc, throws in case of 4xx or 5xx errors, so those error will not be mapped to ResponseEntity with http error code here.